### PR TITLE
Add Markers plugin

### PR DESCRIPTION
**Pull request by osheroff**
This adds a "markers" plugin to wavesurfer, very similar to what Pro Tools gives you.  Just a way to easily jump around in a waveform. 

Lots more customization options are possible, but I figured I'd wait for the community for ask for it. 

Screen Shot 2021-02-24 at 12 38 15 PM

### CODE REVIEW

File: example/zoom/index.html [hunk 1]
```diff
@@ -17,10 +17,11 @@
         <script src="../../dist/wavesurfer.js"></script>
         <script src="../../dist/plugin/wavesurfer.timeline.js"></script>
         <script src="../../dist/plugin/wavesurfer.regions.js"></script>
+        <script src="../../dist/plugin/wavesurfer.markers.js"></script>
```
Comment (thijstriemstra) [inline example/zoom/index.html:20]: i would like to keep the zoom example as clean as possible (and remove other plugins in the future).

File: src/plugin/markers/index.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        this.markerWidth = 11;
+        this.markerHeight = 22;
+
+        this._onBackendCreated = () => {
+            this.wrapper = this.wavesurfer.drawer.wrapper;
+            if (this.params.markers) {
+                this.params.markers.forEach(marker => this.add(marker));
+            }
+            window.addEventListener('resize', this._updateMarkerPositions.bind(this), true);
+            window.addEventListener('orientationchange', this._updateMarkerPositions.bind(this), true);
```
Comment (thijstriemstra) [inline src/plugin/markers/index.js:87]: these event listeners need to be cleanup on destroy.

**Review by thijstriemstra — CHANGES_REQUESTED** [review]
thanks! can you also add a version marker in the new plugin and a changelog entry?

File: example/zoom/index.html (hunk 1 above)
Comment (osheroff) [reply example/zoom/index.html:20]: ok, I'll pull it out.  was really just using it to test that my code reacted to zooming/scrolling properly.

**Comment by osheroff** [comment]
> thanks! can you also add a version marker in the new plugin and a changelog entry?

can you clarify what you mean by version marker?

**Comment by osheroff** [comment]
I've addressed all other issue besides the version marker

**Comment by thijstriemstra** [comment]
> can you clarify what you mean by version marker?

see https://jsdoc.app/tags-since.html, I would put it in the new plugin class. The version nr can be 4.x.0 where x is current version + 1

File: example/annotation/app.js [hunk 3]
```diff
@@ -58,12 +58,9 @@ document.addEventListener('DOMContentLoaded', function() {
             //         wavesurfer.getDuration()
             //     )
             // );
-            wavesurfer.util
-                .ajax({
-                    responseType: 'json',
-                    url: 'annotations.json'
-                })
-                .on('success', function(data) {
+            fetch('/example/annotation/annotations.json')
```
Comment (thijstriemstra) [inline example/annotation/app.js:61]: no need to use an absolute url here, i'd prefer to use the prev url format (unless it somehow doesn't work with such a relative url but i doubt it)

File: src/plugin/markers/index.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+        this.style(labelDiv, {
+            display: "flex",
+            "align-items": "center",
+            cursor: "pointer"
+        });
+
+        el.appendChild(labelDiv);
+
+        labelDiv.addEventListener("click", e => {
```
Comment (thijstriemstra) [inline src/plugin/markers/index.js:235]: this event listener is never cleaned up is it?

File: CHANGES.md [hunk 5]
```diff
@@ -1,12 +1,16 @@
 wavesurfer.js changelog
 =======================
 
+x.x.x (unreleased)
+------------------
+- Markers plugin: add new plugin that allows for timeline markers
```
Comment (thijstriemstra) [inline CHANGES.md:6]: pr nr is missing

File: src/plugin/markers/index.js (hunk 4 above)
Comment (osheroff) [reply src/plugin/markers/index.js:235]: no but the div is removed from the DOM... would think that was adequate but not sure.

Comment (thijstriemstra) [reply src/plugin/markers/index.js:235]: add a console log in the click handler, destroy wavesurfer, create a new one and see if there are 2 click console logs printed..

File: example/annotation/app.js (hunk 3 above)
Comment (osheroff) [reply example/annotation/app.js]: fixed

File: src/plugin/markers/index.js (hunk 4 above)
Comment (osheroff) [reply src/plugin/markers/index.js:236]: confirmed, the handler is removed.

**Comment by osheroff** [comment]
issues addressed.

File: CHANGES.md [hunk 6]
```diff
@@ -1,12 +1,16 @@
 wavesurfer.js changelog
 =======================
 
+x.x.x (unreleased)
+------------------
+- Markers plugin: add new plugin that allows for timeline markers (#2196)
```
Comment (thijstriemstra) [inline CHANGES.md:6]: use 4.6.0 here

**Review by thijstriemstra — CHANGES_REQUESTED** [review]

**Review by thijstriemstra — APPROVED** [review]
thanks!

**Pull request merged**